ci/test/00_setup_env_native_asan.sh

```diff
@@ -28,7 +28,8 @@ export BITCOIN_CONFIG="\
  -DCMAKE_C_COMPILER=clang-18 \
  -DCMAKE_CXX_COMPILER=clang++-18 \
  -DCMAKE_C_FLAGS='-ftrivial-auto-var-init=pattern' \
- -DCMAKE_CXX_FLAGS='-ftrivial-auto-var-init=pattern' \
+ -DCMAKE_CXX_FLAGS='-ftrivial-auto-var-init=pattern -Wno-error=deprecated-declarations' \
```

Comment:
A note for other readers:
`Wno-error=deprecated-declarations` is here because the leveldb subtree [currently uses](https://github.com/bitcoin-core/leveldb-subtree/issues/43) `std::aligned_storage`.  ([deprecated in C++23](https://wg21.link/P1413R3)) 

Patches introducing features deprecated in C++20 or earlier will still be caught by other CI tasks.